gpflow/kernels.py

```diff
@@ -1005,6 +1006,94 @@ def Kdiag(self, X, presliced=False):
         return reduce(tf.multiply, [k.Kdiag(X) for k in self.kernels])
 
 
+class ChangePoints(Combination):
+    """
+    The ChangePoints kernel defines a fixed number of change-points along a 1d
+    input space where different kernels govern different parts of the space.
+
+    The kernel is by multiplication and addition of the base kernels with
+    sigmoidal functions (σ). A single change-point kernel is defined as:
+
+        K₁(x, x') * (1 - σ(x)) * (1 - σ(x')) + K₂(x, x') * σ(x) * σ(x')
+
+    where K₁ is deactivated around the change-point and K₂ is activated. The
+    single change-point version can be found in \citet{ldgtg2014}.
+
+    @incollection{ldgtg2014,
+      author = {Lloyd, James Robert and Duvenaud, David and Grosse, Roger and Tenenbaum, Joshua B. and Ghahramani, Zoubin},
+      title = {Automatic Construction and Natural-language Description of Nonparametric Regression Models},
+      booktitle = {Proceedings of the Twenty-Eighth AAAI Conference on Artificial Intelligence},
+      series = {AAAI'14},
+      year = {2014},
+      pages = {1242--1250},
+      numpages = {9},
+      url = {http://dl.acm.org/citation.cfm?id=2893873.2894066},
+      acmid = {2894066},
+      publisher = {AAAI Press},
+    }
+    """
+    def __init__(self, kernels, locations=0.0, widths=1.0, activation_order=None, name=None):
+        super().__init__(kernels, name=name)
+        if self.input_dim > 1:
+            raise ValueError("ChangePoints kernel only valid for 1d inputs")
+
+        if not isinstance(locations, Iterable):
+            locations = [locations]
+        if not isinstance(widths, Iterable):
+            widths = [widths]
+
+        self.locations = Parameter(locations, transform=None, dtype=settings.float_type)
+        self.widths = Parameter(widths, transform=transforms.positive, dtype=settings.float_type)
+        self.activation_order = list(activation_order or range(len(kernels)))
```

Comment:
Generally it's more robust to make explicit use of the ternary if-else operator (and necessary when dealing with numpy arrays):
```suggestion
        self.activation_order = list(activation_order if activation_order is not None else range(len(kernels)))
```
What would happen if `activation_order` is passed in as a numpy array? Would that be supported?

Reply:
What was your thinking behind including `activation_order` as an explicit argument (instead of simply relying on the order of `kernels` passed in)?

Reply:
The thinking behind activation order is to support the more general case when you want to define a "change window" i.e. k1 changes to k2 then back to k1. `activation_order` allows this by specifying: `ChangePoints([k1, k2], locations=[1., 2.], activation_order=[0, 1, 0])`

My suspicion was that if I did `ChangePoints([k1, k2, k1], locations=[1., 2.])` then trained the kernel, the weights of the two k1 kernels would become decoupled?